src/data_scraper/scraper_binance.py

```diff
@@ -12,77 +12,87 @@
+                                              if col != 'Timestamp (ms)'})
 
             if len(df) == 0:
                 df = temp_df.copy()
             else:
-                df = df.merge(temp_df, how='outer', on=config.MERGE_DATA_ON)
-
-        # Creating general datetime column in seconds
-        df["datetime"] = df[f"{currency_pair}_Open time"]
+                df = df.merge(temp_df, how='outer', on='Timestamp (ms)')
 
-        return df
+        # Getting the merger column
+        df[config.MERGE_DATA_ON] = pd.to_datetime(df['Timestamp (ms)'], unit='ms')
 
-    def load_dataset(self, start_time, end_time):
-        full_dataset_name = f"{self.dataset_name}_{start_time}_{end_time}"
-        if self.dataset_not_stored(full_dataset_name):
-            to_store_dataset = self.get_data(start_time, end_time)
-            to_store_dataset.to_csv(f"{self.dataset_path}/{full_dataset_name}")
-            self.cache_dataset = to_store_dataset
-        else:
-            self.cache_dataset = pd.read_csv(f"{self.dataset_path}/{full_dataset_name}")
+        # Date column to partition by with parquet later
+        df['date'] = df[config.MERGE_DATA_ON].dt.date
 
-        self.cache_dataset[config.MERGE_DATA_ON] = pd.to_datetime(self.cache_dataset[config.MERGE_DATA_ON],
-                                                                  infer_datetime_format=True)
-
-    def dataset_not_stored(self, full_dataset_name):
-        return not os.path.exists(f"{self.dataset_path}/{full_dataset_name}")
+        return df
 
     def get_stored_data(self, start_time, end_time):
-        data =  self.cache_dataset.loc[
-               (self.cache_dataset["datetime"] > start_time) & (
-                           self.cache_dataset["datetime"] <= end_time), :]
-
+        """
+        Returns the exact slice of cached data that is between start and end timestamp.
+        """
+        data = self.cache_data.loc[(self.cache_data["Timestamp (ms)"] > start_time) &
+                                   (self.cache_data["Timestamp (ms)"] <= end_time), :]
         if len(data) == 0:
             raise RuntimeError("No more data")
+
         return data
+
+    def load_from_disk(self, start_time, end_time) -> None:
+        """
+        Load stored data into cache. To actually get the dataframe returned, call get_stored_data().
+        :param start_time: start timestamp
+        :param end_time: end timestamp
+        :return: None
+        """
+        if not utils.dataset_stored(self.dataset_path, start_time, end_time):
+            for chunk_start, chunk_end in time_helpers.slice_timestamps_in_chunks(start_time, end_time):
+                data = self.scrape_data(chunk_start, chunk_end)
+                utils.save_data(data, self.dataset_path)
+        data = pq.read_table(source=self.dataset_path).to_pandas()
+        data = data.groupby(config.MERGE_DATA_ON, as_index=False).last()  # Only takes the last saved data
```

Comment:
Why do you think we should store previous versions? Why no overwriting them?

Reply:
Yeppp, would be better but there's no `overwrite` parameter in parquet so I'd need to implement this manually and again, probably better to prioritize working on other modules and leave this for later